packages/main/src/features/F6NavigationHelper.js

```diff
@@ -0,0 +1,90 @@
+import { registerFeature } from "@ui5/webcomponents-base/dist/FeaturesRegistry.js";
+import { isF6Next, isF6Previous } from "@ui5/webcomponents-base/dist/Keys.js";
+import { getFirstFocusableElement } from "@ui5/webcomponents-base/dist/util/FocusableElements.js";
+
+class F6NavigationHelper {
+	constructor() {
+		this.attachEventListeners();
+		this.selectedGroup = null;
+		this.groups = [];
+		this.cachedGroups = [];
+	}
+
+	attachEventListeners() {
+		document.addEventListener("keydown", async event => {
+			if (isF6Next(event)) {
+				this.eventHandlingPreparation();
+
+				const nextIndex = this.groups.indexOf(this.selectedGroup);
+				let nextElement = null;
+
+				if (nextIndex > -1) {
+					if (nextIndex + 1 >= this.groups.length) {
+						nextElement = this.groups[0];
+					} else {
+						nextElement = this.groups[nextIndex + 1];
+					}
+				} else {
+					nextElement = this.groups[0];
+				}
+
+				const elementToFocus = await getFirstFocusableElement(nextElement.isUI5Element ? nextElement.getDomRef() : nextElement);
+				elementToFocus.focus();
+			}
+
+			if (isF6Previous(event)) {
+				this.eventHandlingPreparation();
+
+				const nextIndex = this.groups.indexOf(this.selectedGroup);
+				let nextElement = null;
+
+				if (nextIndex > -1) {
+					if (nextIndex - 1 < 0) {
+						nextElement = this.groups[this.groups.length - 1];
+					} else {
+						// Handle the situation where the first focusable element of two neighbor groups is the same
+						// For example:
+						// <ui5-flexible-column-layout>
+						//     <ui5-list>
+						//         <ui5-li>List Item</ui5-li>
+						//     </ui5-list>
+						// </ui5-flexible-column-layout>
+						// Here for both FCL & List the firstFoccusableElement is the same (the ui5-li)
+
+						const firstFocusable = await getFirstFocusableElement(this.groups[nextIndex - 1]);
+						const shouldSkipParent = firstFocusable === await getFirstFocusableElement(this.groups[nextIndex]);
```

Comment:
This looks complicated. Wouldn't it be easier to filter out duplicated DOM elements from `this.groups` when you construct them (for example after `.filter` run another `.filter` that removes all that are already in `this.groups`)? Even if we dont  Also, it is possible that there might be 3 or more groups with the same situation and here you skip only 1 level.

Reply:
I don't think that it is a good idea to run this check for each group on the page. In a page built with ui5 web components, we will have 30+ groups and it won't be efficient as this operation will run on **every** keydown.

I agree when it comes to the nesting levels. However, a case with more than 2 levels of nesting is not feasible at this point.